cmd/ui/main.go

```diff
@@ -2,49 +2,51 @@ package main
 
 import (
 	"embed"
-	"encoding/json"
+	"flag"
 	"io/fs"
 	"net/http"
+	"net/http/httputil"
+	"net/url"
 	"os"
 	"path/filepath"
 
 	"github.com/sirupsen/logrus"
 )
 
+var log = logrus.New()
+
 func main() {
-	log := logrus.New()
+	apiUrlFlag := flag.String("api-url", "http://localhost:8000", "The URL for the applications API server")
+
 	mux := http.NewServeMux()
 
 	mux.Handle("/health/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Should this perhaps `HEAD` the `api-url` to check that the api-url is healthy?

Or do we consider that if the proxy is available at all, then the system is healthy, irrespective of it's primary dependency?

Reply:
In my mind, the health check controls routing of traffic in k8s (once healthy). I think the `ui` server is healthy even if the api-url is not. It can return the UI that may render errors if the API is not healthy.

Returning a UI with an error message seems better than not routing traffic to this service.

Reply:
No longer relevant, using an "embedded" API server approach instead of proxying